frontend/packages/integration-tests-cypress/tests/crud/namespace-crud.spec.ts

```diff
@@ -53,54 +83,122 @@ describe('Namespace', () => {
-    projectDropdown.selectProject('default');
-    projectDropdown.shouldContain('default');
-    cy.get(`[data-test-rows="resource-row"]`)
-      .first()
-      .find('a')
-      .first()
-      .click();
-    projectDropdown.shouldContain('default');
+    projectDropdown.selectProject(defaultProjectName);
+    projectDropdown.shouldContain(defaultProjectName);
+    listPage.rows.clickFirstLinkInFirstRow();
+    detailsPage.isLoaded();
+    projectDropdown.shouldContain(defaultProjectName);
     nav.sidenav.clickNavLink(['Workloads', 'Secrets']);
-    projectDropdown.shouldContain('default');
-    cy.get(`[data-test-rows="resource-row"]`)
-      .first()
-      .find('a')
-      .first()
+    projectDropdown.shouldContain(defaultProjectName);
+    listPage.rows.clickFirstLinkInFirstRow();
+    detailsPage.isLoaded();
+    projectDropdown.shouldContain(defaultProjectName);
+    detailsPage
+      .breadcrumb(0)
+      .contains('Secrets')
       .click();
-    projectDropdown.shouldContain('default');
-    cy.byLegacyTestID('breadcrumb-link-0').click();
-    projectDropdown.shouldContain('default');
+    listPage.rows.shouldBeLoaded();
+    projectDropdown.shouldContain(defaultProjectName);
+  });
+
+  ['Roles', 'RoleBindings'].forEach((rolesOrBindings) => {
+    it.only(`tests ${rolesOrBindings} and Cluster${rolesOrBindings} detail page breadcrumbs`, () => {
+      nav.sidenav.clickNavLink(['User Management', rolesOrBindings]);
+
+      cy.log(`test Cluster${rolesOrBindings} & All Projects`);
+      projectDropdown.selectProject(allProjectsDropdownLabel);
+      listPage.rows.shouldBeLoaded();
+      listPage.filter.by('cluster');
+      listPage.filter.numberOfActiveFiltersShouldBe(1);
+      listPage.filter.byName('alertmanager-main');
+      listPage.rows.clickRowByName('alertmanager-main');
+      detailsPage.isLoaded();
+      projectDropdown.shouldNotExist();
+      detailsPage
+        .breadcrumb(0)
+        .contains(rolesOrBindings)
+        .click();
+      listPage.rows.shouldBeLoaded();
+      projectDropdown.shouldContain(allProjectsDropdownLabel);
+      listPage.titleShouldHaveText(rolesOrBindings);
+
+      cy.log(`test Cluster${rolesOrBindings} & default project`);
+      projectDropdown.selectProject(defaultProjectName);
+      projectDropdown.shouldContain(defaultProjectName);
+      listPage.filter.by('cluster');
+      listPage.filter.byName('alertmanager-main');
+      listPage.rows.clickRowByName('alertmanager-main');
+      detailsPage.isLoaded();
+      projectDropdown.shouldNotExist();
+      detailsPage
+        .breadcrumb(0)
+        .contains(rolesOrBindings)
+        .click();
+      listPage.rows.shouldBeLoaded();
+      projectDropdown.shouldContain(defaultProjectName);
+      listPage.titleShouldHaveText(rolesOrBindings);
+
+      cy.log(`test ${rolesOrBindings} & default project`);
+      listPage.filter.by('namespace');
+      listPage.filter.byName('prometheus-k8s');
```

Comment:
Same here. I would not assume this role exists on the cluster (or the hard-coded names below).

Reply:
fixed, created example Role, RoleBinding, ClusterRole, and ClusterRoleBinding instances using test namespace/project.